README.md

```diff
@@ -40,61 +60,197 @@ import { ReactChatbot } from "@vectara/react-chatbot";
   emptyStateDisplay={<MyEmptyStateDisplayComponent />}
   isInitiallyOpen={false}
+  zIndex={ /* (optional) number representing the z-index the component should have */ }
 />;
 ```
 
-### Configuration options
+#### <u>Configuration Options</u>
 
-#### `customerId` (required)
+##### `customerId` (required)
 
 Every Vectara account is associated with a customer ID. You can find your customer ID by logging into the [Vectara Console](https://console.vectara.com/) and opening your account dropdown in the top-right corner.
 
-#### `corpusIds` (required)
+##### `corpusIds` (required)
 
 After you [create a corpus](https://docs.vectara.com/docs/console-ui/creating-a-corpus), you can find its ID by navigating to the corpus and looking in the top-left corner, next to the corpus name.
 
-#### `apiKey` (required)
+##### `apiKey` (required)
 
 API keys enable applications to access data inside of corpora. Learn how to [create a **QueryService** API key](https://docs.vectara.com/docs/console-ui/manage-api-access#create-an-api-key).
 
-#### `title` (optional)
+##### `title` (optional)
 
 Configure the title in the header of the chatbot window.
 
-#### `placeholder` (optional)
+##### `placeholder` (optional)
 
 Configure the placeholder text in the chatbot's input.
 
-#### `emptyStateDisplay` (optional)
+##### `emptyStateDisplay` (optional)
 
 Configure JSX content to render in the messages window when there are no messages to display.
 
-#### `isInitiallyOpen` (optional)
+##### `isInitiallyOpen` (optional)
 
 Set the chat window to be opened on initial render.
 
-#### `inputSize` (optional)
+##### `inputSize` (optional)
+
+Control the size of the input - either "large" (18px) or "medium" (14px)
+
+##### `zIndex` (optional)
+
+Customize the z-index of the chatbot widget
+
+### Use your own views with the useChat hook
+
+Install React-Chatbot:
+
+```shell
+npm install --save @vectara/react-chatbot
+```
+
+Then use the `useChat` hook in your application like this:
+
+```js
+import { useChat } from "@vectara/react-chatbot/lib";
+
+/* snip */
+
+const { sendMessage, messageHistory, isLoading, hasError } = useChat(
+  "CUSTOMER_ID",
+  ["CORPUS_ID_1", "CORPUS_ID_2", "CORPUS_ID_N"],
+  "API_KEY"
+);
+```
+
+The values returned by the hook can be passed on to your custom components as props or used in any way you wish.
+
+#### <u>Hook Values</u>
+
+##### sendMessage: `async ({ query: string; isRetry?: boolean }) => void;`
+
+This is used to send a message to the chat API. Send `true` for the optional `isRetry` flag to if this is retrying a previously failed message. This allows the internal logic to correctly link the next successful answer to the failed query.
 
-Used to control the size of the input - either "large" (18px) or "medium" (14px)
+##### messageHistory: `ChatTurn[]`
+
+This is an array of objects representing question and answer pairs from the entire conversation. Each pair is a `ChatTurn` object, which is typed as:
+
+```
+export type ChatTurn = {
```

Comment:
WDYT of linking to https://github.com/vectara/react-chatbot/blob/main/src/types.ts instead of duplicating the definition here? It will make maintenance easier by reducing the risk of divergence.

Reply:
https://github.com/vectara/react-chatbot/pull/13/commits/908d9f7b3f5ab2e755606cd271302ce1306b7c1e